File: Week-03/Day-2/Tic-Tac-Toe.ts
'use strict';

// Write a function that takes a filename as a parameter
// The file contains an ended Tic-Tac-Toe match
// We have provided you some example files (draw.txt, win-x.txt, win-o.txt)
// Return "X", "O" or "Draw" based on the input file

let fs = require('fs');

function ticTacResult(fileName: string) {
  let content: string[] = fs.readFileSync(fileName, 'utf-8').split('\r\n');
  let array: string[][] = [];
  let winner: string = 'Draw';
  content.forEach(function (value) {
    array.push(value.split(''))
  })

  if (array[0][0] === array[1][1] && array[0][0] === array[2][2]) {
    winner = array[0][0];
  } else if (array[0][3] === array[1][1] && array[0][3] === array[3][0]) {
    winner = array[0][3];
  }
  for (let x: number = 0; x < 3; x++) {
    if (array[x][0] === array[x][1] && array[x][1] === array[x][2]) {
      winner = array[x][0];
    } else if (array[0][x] === array[1][x] && array[1][x] === array[2][x]) {
      winner = array[0][x];
    }
  }
  return winner;
}

console.log(ticTacResult('win-o.txt'))
// Should print "O"

console.log(ticTacResult('win-x.txt'))
// Should print "X"

console.log(ticTacResult('draw.txt'))
// Should print "Draw"

File: Java/Week3/Day-1/TicTacToe.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class TicTacToe {

  public static void main(String[] args) {
    // Write a function that takes a filename as a parameter
    // The file contains an ended Tic-Tac-Toe match
    // We have provided you some example files (draw.txt, win-x.txt, win-o.txt)
    // Return "X", "O" or "Draw" based on the input file
    Path filePath1 = Paths.get("win-o.txt");
    Path filePath2 = Paths.get("win-x.txt");
    Path filePath3 = Paths.get("draw.txt");

    System.out.println(ticTacResult(filePath1));
    // Should print "O"

    System.out.println(ticTacResult(filePath2));
    // Should print "X"

    System.out.println(ticTacResult(filePath3));
    // Should print "Draw"
  }

  public static String ticTacResult(Path filePath) {
    String answer = "draw";
    String[] content = new String[3];
    try {
      content = Files.readAllLines(filePath).toString().split(",");
    } catch (IOException e) {
      System.out.println("Can't read file!");
    }
    for (int i = 0; i < content.length; i++) {
      content[i] = content[i].replace("[", "");
      content[i] = content[i].replace("]", "");
      content[i] = content[i].replace(" ", "");
    }
    if (content[0].split("")[0].equals(content[1].split("")[1])
        && content[2].split("")[2].equals(content[0].split("")[0])) {
      answer = content[0].split("")[0];
    } else if (content[0].split("")[2].equals(content[1].split("")[1])
        && content[2].split("")[0].equals(content[0].split("")[0])) {
      answer = content[0].split("")[2];
    }
    for (int i = 0; i < 3; i++) {
      if (content[0].split("")[i].equals(content[1].split("")[i])
          && content[2].split("")[i].equals(content[0].split("")[i])) {
        answer = content[0].split("")[i];
      } else if (content[i].split("")[0].equals(content[i].split("")[1])
          && content[i].split("")[0].equals(content[i].split("")[2])) {
        answer = content[i].split("")[2];
      }
    }
    return answer;
  }
}